### PR TITLE
mag3110: change method name/signature for obtaining temperature to be consistent

### PR DESCRIPTION
**Pull request by deadprogram**
This PR changes the method name/signature for obtaining temperature to be uniform with other similar drivers.

### CODE REVIEW

File: mag3110/mag3110.go [hunk 1]
```diff
@@ -49,8 +49,8 @@ func (d Device) ReadMagnetic() (x int16, y int16, z int16) {
 }
 
 // ReadTemperature reads the current die temperature in degrees Celsius.
-func (d Device) ReadTemperature() (temp int8) {
+func (d Device) ReadTemperature() (int32, error) {
 	data := make([]byte, 1)
 	d.bus.ReadRegister(uint8(d.Address), DIE_TEMP, data)
-	return int8(data[0])
+	return int32(data[0]), nil
```
Comment (aykevl) [inline mag3110/mag3110.go]: This is in units of 1°C, not in units of 0.001°C as described in https://github.com/tinygo-org/drivers/wiki/Driver-design---API. While the signature may be the same, the interface works differently.

Other than that, this is a good change.

Comment (deadprogram) [reply mag3110/mag3110.go]: OK updated so the functionality matches by returning the correct scaling of the result.

**Comment by deadprogram** [comment]
Ready for review.

**Comment by deadprogram** [comment]
Reminder this is ready for review.

**Comment by aykevl** [comment]
Merged in b7fe897b8ceaea823ac79509a20bdca3182d9737. Thank you!

**Comment by deadprogram** [comment]
OK, then closing since that code was merged.

**Comment by aykevl** [comment]
Ah yes, forgot to close.

**Pull request closed without merging**